src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/RazorCompletionFactsService.cs

```diff
@@ -48,4 +56,40 @@ public ImmutableArray<RazorCompletionItem> GetCompletionItems(RazorCompletionCon
 
         return completions.DrainToImmutable();
     }
+
+    // Internal for testing
+    [return: NotNullIfNotNull(nameof(originalNode))]
+    internal static SyntaxNode? AdjustSyntaxNodeForWordBoundary(SyntaxNode? originalNode, int requestIndex, HtmlFactsService htmlFactsService)
+    {
+        if (originalNode == null)
+        {
+            return null;
+        }
+
+        // If we're on a word boundary, ie: <a hr| />, then with `includeWhitespace`, we'll get back back a whitespace node.
```

Comment:
Thank you for the good comments! :)

Reply:
They're how I rationalized the behavior to myself :laughing: